server/handlers.go

```diff
@@ -1099,10 +1098,22 @@ func (s *Server) RedirectHandler(h http.Handler) http.HandlerFunc {
 		} else if strings.HasSuffix(ipAddrFromRemoteAddr(r.Host), ".onion") {
 			// .onion addresses cannot get a valid certificate, so don't redirect
 		} else if r.Header.Get("X-Forwarded-Proto") == "https" {
-		} else if r.URL.Scheme == "https" {
+		} else if r.TLS != nil {
 		} else {
 			u := getURL(r, s.proxyPort)
 			u.Scheme = "https"
+			if len(s.proxyPort) == 0 && len(s.TLSListenerString) > 0 {
+				_, port, err := net.SplitHostPort(s.TLSListenerString)
```

Comment:
Should we assign host here and use it further down the same way as we do in the first block of code above?

Reply:
indeed not. in the first block of code we use the `request.Host`
here we use the `TLSListenerString`, that's in the format `x.x.x.x:port` or even only `:port`

we should use only the port of the tls listener, and only in the case a proxy port is not set, and `u.Hostname()` for the host (that will be the one coming from the first block/request)

Reply:
True. In my case i had it setup to actually have the hostname:port configuration, but this is certainly more flexible.